index.js

```diff
@@ -51,10 +51,10 @@ const getCityState = async (userZip) => {
 //get weather data from openweathermap.org based on zip code
 async function fetchWeatherData(cityStateObj) {
     let response;
-    let city = cityStateObj.city.toLowerCase()
-    let state = cityStateObj.state.toLowerCase()
 
     try {
+        let city = cityStateObj.city?.toLowerCase()
+        let state = cityStateObj.state?.toLowerCase()
 
         response = await axios(`https://api.openweathermap.org/data/2.5/weather?q=${city},${state},usa&units=imperial&APPID=${OPEN_WEATHER_KEY}`);
```

Comment:
`fetch` is built in and as performant as possible. It changes how the response is processed though and I can't make suggestions for code in GitHub outside of the areas that were changed. I'd have to create a separate pull request, but the gist is line 62 would be:

```javascript
        const { data: { weather: [{ description }], main: { temp: temperature } } } = (
          await response.json()
        )
```
Your return statement then would be:

```javascript
        return { city, state, temperature, description }
```

```suggestion
        response = await fetch(`https://api.openweathermap.org/data/2.5/weather?q=${combo},usa&units=imperial&APPID=${OPEN_WEATHER_KEY}`);
```